pyproject.toml

```diff
@@ -99,14 +99,19 @@ env = { TEMPORAL_INTEGRATION_TEST = "1" }
 cmd = "pip uninstall temporalio -y"
 
 [tool.pytest.ini_options]
-addopts = "-p no:warnings"
 asyncio_mode = "auto"
 log_cli = true
 log_cli_level = "INFO"
 log_cli_format = "%(asctime)s [%(levelname)8s] %(message)s (%(filename)s:%(lineno)s)"
 testpaths = ["tests"]
 timeout = 600
 timeout_func_only = true
+filterwarnings = [
+  "error::temporalio.worker._workflow_instance.UnfinishedUpdateHandlerWarning",
```

Comment:
Seeing this, I think we need `category="UnfinishedUpdateHandlerWarning"` or something when calling `warn`. I didn't read https://docs.python.org/3/library/warnings.html#warnings.warn close enough to realize they are using the fully qualified class name which I don't think we want or if we do want, we should move these classes to the `temporalio.workflow` package (but I think just changing category is easiest). Could also be `category=f"TemporalUnfinishedUpdateHandlerWarning"` if we're worried about global clash.

Reply:
Moved the warning classes to `temporalio.workflow` fa68bbcecfd57dc71d5f207dc0272d19c3005753 so that we can carry on calling `warnings.warn(MyWarningCls(message))` and users see `temporalio.workflow.UnfinishedUpdateHandlerWarning` and would use that to ignore/raise the warnings.